### ISSUE TITLE
Merging two LSTMS throwing an error 

### ISSUE THREAD

**Issue by pchowdhry** [fchollet/keras#4665]
I’m working with Keras on a sentence similarity task (using the STS dataset) and am having problems merging the layers. The data consists of 1184 sentence pairs each scored between 0 and 5. Below are the shapes of my numpy arrays. I’ve padded each of the sentences to 50 words and run them through and embedding layer, using the glove embedding’s with 100 dimensions. When merging the two networks I'm getting the following error... Apologize in advance as I'm pretty sure this is user error...

`Exception: Error when checking model input: the list of Numpy arrays that you are passing to your model is not the size the model expected. Expected to see 1 arrays but instead got the following list of 2 arrays:`

Here is my code (left out the mundane data prep, but included my numpy shapes)

```
total training data = 1184
X1.shape = (1184, 50)
X2.shape = (1184, 50)
Y.shape = (1184, 1)


embedding_matrix = np.zeros((len(word_index) + 1, EMBEDDING_DIM))
for word, i in word_index.items():
    embedding_vector = embeddings_index.get(word)
    if embedding_vector is not None:
        # words not found in embedding index will be all-zeros.
        embedding_matrix[i] = embedding_vector

embedding_layer = Embedding(len(word_index) + 1,
                            EMBEDDING_DIM,
                            weights=[embedding_matrix],
                            input_length=50,
                            trainable=False)

s1rnn = Sequential()
s1rnn.add(embedding_layer)
s1rnn.add(LSTM(128, input_shape=(100, 1)))
s1rnn.add(Dense(1))

s2rnn = Sequential()
s2rnn.add(embedding_layer)
s2rnn.add(LSTM(128, input_shape=(100, 1)))
s2rnn.add(Dense(1))

model = Sequential()
model.add(Merge([s1rnn,s2rnn],mode='concat'))
model.add(Dense(1))
model.compile(loss='mean_squared_error', optimizer='RMSprop', metrics=['accuracy'])
model.fit([X1,X2], Y,batch_size=32, nb_epoch=100, validation_split=0.05)
```

Thanks in advance

**Comment by bkj** [comment]
Seems like it doesn't like the shared `embedding_layer` -- haven't tested, but the `input_shape` becomes correct if you change to (roughly)

```
s1rnn = Sequential()
s1rnn.add(Embedding(...))
s1rnn.add(LSTM(...))
s1rnn.add(Dense(...))

s2rnn = Sequential()
s2rnn.add(Embedding(...))
s2rnn.add(LSTM(...))
s2rnn.add(Dense(...))

model = Sequential()
model.add(Merge([s1rnn, s2rnn], mode='concat', concat_axis=1))
model.add(Dense(1))

model.input_shape # => `[(None, 50), (None, 50)]`
```

FWIW, for this kind of thing, I usually use the functional API.  Also BTW, are you sure you want two separate LSTMs? Or do you want to tie the weights on those too?

**Comment by pchowdhry** [comment]
Hi Ben, thanks a lot for looking at this, AND, crap! I didn't even think it could be the shared embedding layer. I just copied it into two separate embedding layers and everything worked. I think I want to share the weights... I'm using this as a learning task, varying items like optimizers and activation functions to get a feel for the software. I've got another sample where I've concatenated the sentences with stop tokens, etc.. I'm also going to try the BiLSTM, etc and a couple of other architectures.
